app/controllers/api/v1/vocabularies_controller.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+###
+# @description: Place all the actions related to vocabularies
+###
+class Api::V1::VocabulariesController < ApplicationController
+  before_action :authorize_with_policy
+
+  include Pundit
+
+  ###
+  # @description: Returns the list of vocabularies for the logged in
+  #   user's organization
+  ###
+  def index
+    vocabularies = Vocabulary.all.order(name: :desc)
+    render json: vocabularies
+  end
+
+  ###
+  # @description: Creates a vocabulary
+  ###
+  def create
+    @vocabulary = Vocabulary.create!(valid_params)
+
+    render json: {
+      success: true,
+      vocabulary: @vocabulary
+    }
+  end
+
+  private
+
+  ###
+  # @description: Execute the authorization policy
+  ###
+  def authorize_with_policy
+    authorize vocabulary
+  end
+
+  ###
+  # @description: Get the current model record
+  # @return [ActiveRecord]
+  ###
+  def vocabulary
+    @vocabulary = params[:id].present? ? Vocabulary.find(params[:id]) : (Vocabulary.first || Vocabulary)
```

Comment:
`find` should be scoped to the current organization. Also, returning a constant when there are no records doesn't look right. I think `404` would make perfect sense in such a case.

Reply:
@excelsior the Pundit gem needs an instance of the model in order to perform the authorization policy. It is, by executing the method `authorize`.

So this method `vocabulary` is just meant to find the right model (when the route is for listing a specific vocabulary) or a random one.

To find a vocabulary, the method `show` would be just fine (as in the rest of the controllers), but in this case we don't need that, so that method doesn't exist.